backend/cli_commands/SyncCommand.py

```diff
@@ -87,6 +85,63 @@ def sync_files(mission_path, mission):
                         )
                     except Exception as e:
                         logging.error(f"Error processing {mcap_path}: {e}")
+                else:
+                    file = File.objects.get(file=mcap_path)
+                # Get all video files in this folder
+                videos_in_folder = {
+                    video: os.path.join(subfolder_path, video)
+                    for video in storage.listdir(subfolder_path)[1]
+                    if video.endswith(".mp4")
+                }
+                logging.info(
+                    f"Found {len(videos_in_folder)} video files in folder {subfolder_path}"
+                )
+                # Process each topic in the metadata
+                for topic_name, topic_data in metadata.items():
+                    # Try to find a corresponding video file
+                    matching_video = None
+
+                    for video_filename in videos_in_folder.keys():
+                        # Match by topic name (adjust this logic if needed)
+                        if (
+                            topic_name.replace("/", "-") in video_filename
+                        ):  # Handle path-based topic names
+                            matching_video = videos_in_folder[video_filename]
+                            break
```

Comment:
```suggestion
                    matching_video = videos_in_folder.get(
                        topic_name.replace("/", "-") + ".mp4", None
                    )
```
You are already using a dict then make good use of it